lib/events.js

```diff
@@ -158,28 +167,33 @@ EventEmitter.prototype.emit = function emit(type) {
     needDomainExit = true;
   }
 
-  var isFn = typeof handler === 'function';
+  isFn = typeof handler === 'function';
+  if (!isFn && typeof handler.listener === 'function') {
```

Comment:
elsewhere the code just checks if the thing is an object and doesn't make these additional `function` checks. That should likely be made more consistent.